### DISCUSSION TITLE
Dynamic routing on server side

### DISCUSSION THREAD

**Discussion by Madd0g** [ElMassimo/iles#37] [Q&A] [answered]
If I understand correctly, iles only does SSG, not SSR? So "full" dynamic routing with logic running on the server side is not possible?

That's why `getStaticPaths` has to be specified ahead of time? 

Are there plans to support having a server side? Maybe running iles as a middleware of something?

Thanks

**Comment by ElMassimo** [comment, marked as answer]
Correct, currently îles only performs static generation, there is no server.

In the future I might explore adding server-side rendering support via `vite-plugin-ssr`.

The API that currently allows a page to receive props, as well as the file-system routing would remain backwards compatible.
